shopinvader/models/product_product.py

```diff
@@ -59,14 +59,17 @@ def _add_to_cart_allowed(self, backend, partner=None):
             self.shopinvader_bind_ids.filtered(lambda x: x.backend_id == backend)
         )
 
-    def _get_invader_variant(self, backend, lang):
+    def _get_invader_variant(self, backend, lang, only_active=True):
         """Retrieve invader variant by backend and language.
 
         :param backend: backend recordset
         :param lang: lang code
         """
+        active = (True,) if only_active else (True, False)
         return self.shopinvader_bind_ids.filtered(
-            lambda x: x.backend_id == backend and x.lang_id.code == lang
+            lambda x: x.backend_id == backend
+            and x.lang_id.code == lang
+            and x.active in active
         )
```

Comment:
suppose the change should be done here
https://github.com/shopinvader/odoo-shopinvader/blob/6898c93dcc1af3419c4a3b517a4abb03492a5593/shopinvader_wishlist/models/product_set.py#L52

Reply:
The idea is to make this behavior the default when getting a shopinvader variant, not just for the wishlist.

@simahawk we checked this together, but maybe it's better to follow Iryna's approach to make sure this doesn't break any other implementations?

Reply:
@JuMiSanAr she's right in the sense that if you want to be able to filter explicitly on the active flag, you'll have to use `active_test=False` before you load the bindings otherwise inactive records won't be considered.

Reply:
Sorry, missing one point: this method is not only for the wishlist case. Hence, to let the flag fully control the behavior of the method we should probably set active_test in the context based on the value of the active_only flag.